docs: example of router-store standalone api setup is incorrect

### Information

The usage for `routerReducer` should be like `provideStore({ router: routerReducer })`, instead of `provideState({ router: routerReducer })` in the current documetation:

```diff
import { bootstrapApplication } from '@angular/platform-browser';
import { provideRouter } from '@angular/router';
import { provideStore, provideState } from '@ngrx/store';
import { provideRouterStore, routerReducer } from '@ngrx/router-store';

import { AppComponent } from './app.component';

bootstrapApplication(AppComponent, {
  providers: [
    provideRouter([
      // routes
    ]),
-    provideStore(),
-    provideState({
-      router: routerReducer,
-    }),
+    provideStore({
+      router: routerReducer,
+    }),
    provideRouterStore()
  ],
});
```

### Documentation page

https://ngrx.io/guide/router-store#using-the-standalone-api

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

Hi @brandonroberts, thanks for your reply.

But when I trying out the example code, it doesn't work, and the following error will show up:

```
Error: src/main.ts:16:7 - error TS2345: Argument of type '{ router: <RouterState extends BaseRouterStoreState = SerializedRouterStateSnapshot, Result = RouterReducerState<RouterState>>(state: Result | undefined, action: Action) => Result; }' is not assignable to parameter of type 'FeatureSlice<unknown, Action>'.
  Object literal may only specify known properties, and 'router' does not exist in type 'FeatureSlice<unknown, Action>'.

16       router: routerReducer,
```

Here is the minimal reproduce: https://github.com/jonz94/ngrx-router-store-issue-3762

Then, I hover on the `provideRouterStore()` in vscode, it shows the following example code, which is different from the one on the doc site:

<img width="961" alt="Screenshot 2023-01-31 at 20 51 57" src="https://user-images.githubusercontent.com/16042676/215765294-ca7204dc-1fe5-4519-a9fd-7c70968dcdc6.png">

After changing the code, it works as expected without any errors.

I also found other examples by searching in this repository:

https://github.com/ngrx/platform/blob/e8cf02b4748ea6a3010c9b7db8818e8b666bd73b/modules/effects/migrations/15_0_0-beta/index.spec.ts#L43

https://github.com/ngrx/platform/blob/e8cf02b4748ea6a3010c9b7db8818e8b666bd73b/projects/standalone-app/src/main.ts#L18

This made me believe that the example code on the doc site might be incorrect in the first place.

I'm sorry I didn't make it clear when I opened the issue.
Please take another look. Thanks!

--- a/projects/ngrx.io/content/guide/router-store/configuration.md
+++ b/projects/ngrx.io/content/guide/router-store/configuration.md
@@ -90,7 +90,7 @@ export class AppModule {}
 <code-example header="main.ts">
 import { bootstrapApplication } from '@angular/platform-browser';
 import { provideRouter } from '@angular/router';
-import { provideStore, provideState } from '@ngrx/store';
+import { provideStore } from '@ngrx/store';
 import { provideRouterStore, routerReducer } from '@ngrx/router-store';
 
 import { AppComponent } from './app.component';
@@ -101,8 +101,7 @@ bootstrapApplication(AppComponent, {
     provideRouter([
       // routes
     ]),
-    provideStore(),
-    provideState({
+    provideStore({
       router: routerReducer,
     }),
     provideRouterStore({

--- a/projects/ngrx.io/content/guide/router-store/index.md
+++ b/projects/ngrx.io/content/guide/router-store/index.md
@@ -36,7 +36,7 @@ Registering the router bindings can also be done using the standalone APIs if yo
 <code-example header="main.ts">
 import { bootstrapApplication } from '@angular/platform-browser';
 import { provideRouter } from '@angular/router';
-import { provideStore, provideState } from '@ngrx/store';
+import { provideStore } from '@ngrx/store';
 import { provideRouterStore, routerReducer } from '@ngrx/router-store';
 
 import { AppComponent } from './app.component';

--- a/projects/ngrx.io/content/guide/router-store/index.md
+++ b/projects/ngrx.io/content/guide/router-store/index.md
@@ -46,8 +46,7 @@ bootstrapApplication(AppComponent, {
     provideRouter([
       // routes
     ]),
-    provideStore(),
-    provideState({
+    provideStore({
       router: routerReducer,
     }),
     provideRouterStore()

